Problem building version 4.81. with HDF5 (missing fuzz directory in 'make dist' tarball)

Hello I am trying to build netcdf version 4.8.1 with HDF5 support.
First I build and checked hdf5 version 1.13.0 together with the support for ZLIB and SZIP 

I am using MSVC 2017 and CMAKE GUI.
I am getting the following error message. 

 Checking for HDF5 version 1.10.6 or later: ON
CMake Error at CMakeLists.txt:840 (MESSAGE):
  HDF5 was built without zlib.  Rebuild HDF5 with zlib.

besides the fact that I can promise that I have built the library with ZLIB, what else can I do ?
best
jac


To report a non-security related issue, please provide:

* the version of the software with which you are encountering an issue
* environmental information (i.e. Operating System, compiler info, java version, python version, etc.)
* a description of the issue with the steps needed to reproduce it

If you have a general question about the software, please view our [Suggested Support Process](https://www.unidata.ucar.edu/support/#process).

Hello 
I am now using version 1.12 of HDF5 but I get an error about a missing fuzz directory (same error msg as for version 1.13 of HDF5)
here are the diags from CMAKE

Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.19044.
Found HDF5 libraries version 1.12.1
Using HDF5 include dir: C:/wrf_hydro/hdf/build/_CPack_Packages/win64/ZIP/HDF5-1.12.1-win64/include;
HDF5 has szip.
-- Checking for HDF5 version 1.10.6 or later: ON
Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR) 
Enabling use of fill value when NC_ERANGE
Enabling a more relaxed check for NC_EINVALCOORDS
CMake Warning at CMakeLists.txt:1599 (MESSAGE):
  mmap not supported under visual studio: disabling MMAP support.


Found m4: C:/Flex Windows/Bison/bin/m4.exe
Ignoring C:/Windows/System32/bash.exe
CMAKE_SOURCE_DIR=C:/wrf_hydro/netcdf/netcdf-c-4.8.1
CMAKE_BINARY_DIR=C:/wrf_hydro/netcdf/netcdf-c-4.8.1/build
CMAKE_CURRENT_SOURCE_DIR=C:/wrf_hydro/netcdf/netcdf-c-4.8.1/nc_test
CMAKE_CURRENT_BINARY_DIR=C:/wrf_hydro/netcdf/netcdf-c-4.8.1/build/nc_test


Configuration Summary:

Building Shared Libraries:     ON
Building netCDF-4:             ON
Building DAP2 Support:         OFF
Building DAP4 Support:         OFF
Building Byte-range Support:   OFF
Building Utilities:            ON

Tests Enabled:              ON
DAP Remote Tests:  ON
Extra Tests:       OFF
Coverage Tests:    
Parallel Tests:    OFF
Large File Tests:  
Extreme Numbers:   ON
Unit Tests:        OFF

Compiler:

Build Type:           DEBUG
CMAKE_C_COMPILER:     C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
CMAKE_C_FLAGS:        /DWIN32 /D_WINDOWS /W3
CMAKE_C_FLAGS_DEBUG:  /MDd /Zi /Ob0 /Od /RTC1
Linking against:      hdf5::hdf5-shared;hdf5::hdf5_hl-shared;C:/wrf_hydro/bin/Debug/libsz_D.lib

# NetCDF C Configuration Summary
==============================

# General
-------
NetCDF Version:		4.8.1
Dispatch Version:       3
Configured On:		
Host System:		--
Build Directory: 	C:/wrf_hydro/netcdf/netcdf-c-4.8.1/build
Install Prefix:         C:/Program Files/netCDF

# Compiling Options
-----------------
C Compiler:		C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
CFLAGS:			/DWIN32 /D_WINDOWS /W3 /MDd /Zi /Ob0 /Od /RTC1
CPPFLAGS:		 
LDFLAGS:		/machine:x64 /LARGEADDRESSAWARE /STACK:40000000 /debug /INCREMENTAL
AM_CFLAGS:		
AM_CPPFLAGS:		
AM_LDFLAGS:		
Shared Library:		yes
Static Library:		no
Extra libraries:	-lhdf5 -lhdf5_hl -lsz_D

# Features
--------
NetCDF-2 API:		yes
HDF4 Support:		no
HDF5 Support:		yes
NetCDF-4 API:		yes
NC-4 Parallel Support:	no
PnetCDF Support:	no
DAP2 Support:		no
DAP4 Support:		no
Byte-Range Support:	no
Diskless Support:	yes
MMap Support:		no
JNA Support:		no
CDF5 Support:		yes
ERANGE Fill Support:	yes
Relaxed Boundary Check:	yes
SZIP Support:           yes
SZIP Write Support:     yes
Parallel Filters:       yes
NCZarr Support:		no
Multi-Filter Support:	no

CMake Error at CMakeLists.txt:2257 (add_subdirectory):
  add_subdirectory given source "fuzz" which is not an existing directory.


Configuring incomplete, errors occurred!
See also "C:/wrf_hydro/netcdf/netcdf-c-4.8.1/build/CMakeFiles/CMakeOutput.log".
See also "C:/wrf_hydro/netcdf/netcdf-c-4.8.1/build/CMakeFiles/CMakeError.log".

thanks 
jac

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2460,7 +2460,9 @@ IF(ENABLE_TESTS)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ncdap_test/pingurl.c ${CMAKE_CURRENT_BINARY_DIR}/dap4_test/pingurl4.c @ONLY NEWLINE_STYLE LF)
 ENDIF()
 
-add_subdirectory(fuzz)
+if(DEFINED ENV{LIB_FUZZING_ENGINE})
+  add_subdirectory(fuzz)
+endif(DEFINED ENV{LIB_FUZZING_ENGINE})
 
 ####
 # Export files

--- a/Makefile.am
+++ b/Makefile.am
@@ -16,7 +16,7 @@ cmake CMakeLists.txt COMPILE.cmake.txt config.h.cmake.in		\
 cmake_uninstall.cmake.in FixBundle.cmake.in nc-config.cmake.in		\
 RELEASE_NOTES.md CTestCustom.cmake CTestConfig.cmake.in			\
 libnetcdf.settings.in netCDFConfig.cmake.in CMakeInstallation.cmake	\
-test-driver-verbose test_common.in
+test-driver-verbose test_common.in fuzz
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = netcdf.pc

--- a/fuzz/CMakeLists.txt
+++ b/fuzz/CMakeLists.txt
@@ -1,3 +1,11 @@
+# This is part of the netCDF package.
+# Copyright 2022 University Corporation for Atmospheric Research/Unidata
+# See COPYRIGHT file for conditions of use.
+#
+# Added for fuzzing in https://github.com/Unidata/netcdf-c/pull/1972
+#
+# @authors Ward Fisher
+
 if(DEFINED ENV{LIB_FUZZING_ENGINE})
   add_executable(fuzz_open fuzz_open.c)
   target_link_libraries(fuzz_open netcdf $ENV{LIB_FUZZING_ENGINE})

--- a/fuzz/Makefile.am
+++ b/fuzz/Makefile.am
@@ -1,0 +1,15 @@
+# This is part of the netCDF package.
+# Copyright 2022 University Corporation for Atmospheric Research/Unidata
+# See COPYRIGHT file for conditions of use.
+#
+# @authors Ward Fisher
+
+# Set AM_CPPFLAGS and AM_LDFLAGS based on user choices.
+include $(top_srcdir)/lib_flags.am
+
+#if(DEFINED ENV{LIB_FUZZING_ENGINE})
+#  add_executable(fuzz_open fuzz_open.c)
+#  target_link_libraries(fuzz_open netcdf $ENV{LIB_FUZZING_ENGINE})
+#endif()
+
+EXTRA_DIST = fuzz.c fuzz_open.c

--- a/libnczarr/Makefile.am
+++ b/libnczarr/Makefile.am
@@ -55,7 +55,6 @@ zarr.h \
 zcache.h \
 zchunking.h \
 zdispatch.h \
-ztracedispatch.h \
 zincludes.h \
 zinternal.h \
 zmap.h \

--- a/nc_test4/Makefile.am
+++ b/nc_test4/Makefile.am
@@ -104,7 +104,8 @@ ref_filteredvv.cdl ref_multi.cdl                                        \
 ref_ncgenF.cdl ref_nccopyF.cdl                                          \
 ref_filter_repeat.txt ref_fillonly.cdl test_fillonly.sh                 \
 ref_filter_order_create.txt ref_filter_order_read.txt			\
-ref_any.cdl tst_specific_filters.sh
+ref_any.cdl tst_specific_filters.sh \
+tst_virtual_datasets.c
 
 CLEANFILES = tst_mpi_parallel.bin cdm_sea_soundings.nc bm_chunking.nc	\
 tst_floats_1D.cdl floats_1D_3.nc floats_1D.cdl tst_*.nc tmp_*.txt       \

--- a/ncdump/cdl/Makefile.am
+++ b/ncdump/cdl/Makefile.am
@@ -25,6 +25,7 @@ ref_solar.cdl unlimtest1.cdl unlimtest2.cdl \
 ref_tst_opaque_data.cdl  \
 ref_tst_vlen_data.cdl ref_tst_vlen_data2.cdl \
 ref_niltest.cdl ref_tst_h_scalar.cdl ref_tst_econst.cdl \
+ref_tst_econst2.cdl \
 ref_tst_nul3.cdl ref_tst_nul4.cdl ref_tst_names.cdl \
 ref_tst_long_charconst.cdl tst_chararray.cdl ref_keyword.cdl \
 CMakeLists.txt

--- a/ncdump/expected/Makefile.am
+++ b/ncdump/expected/Makefile.am
@@ -19,6 +19,7 @@ n3time.dmp ref_tst_special_atts3.dmp ref_tst_chardata.dmp ref_tst_unlim2.dmp \
 ref_solar.dmp unlimtest1.dmp unlimtest2.dmp \
 ref_tst_vlen_data.dmp ref_tst_vlen_data2.dmp \
 ref_niltest.dmp ref_tst_h_scalar.dmp ref_tst_econst.dmp \
+ref_tst_econst2.dmp \
 ref_tst_nul3.dmp ref_tst_nul4.dmp ref_tst_names.dmp \
 ref_tst_long_charconst.dmp tst_chararray.dmp ref_keyword.dmp \
 CMakeLists.txt

--- a/nczarr_test/Makefile.am
+++ b/nczarr_test/Makefile.am
@@ -151,7 +151,7 @@ ref_groups.h5 ref_byte.zarr.zip ref_byte_fill_value_null.zarr.zip \
 ref_groups_regular.cdl ref_byte.cdl ref_byte_fill_value_null.cdl
 
 # Interoperability files
-EXTRA_DIST += ref_power_901_constants.zip ref_power_901_constants.cdl ref_quotes.zip ref_quotes.cdl ref_zarr_test_data.cdl.gz
+EXTRA_DIST += ref_power_901_constants.zip ref_power_901_constants.cdl ref_quotes.zip ref_quotes.cdl 
 
 CLEANFILES = ut_*.txt ut*.cdl tmp*.nc tmp*.cdl tmp*.txt tmp*.dmp tmp*.zip tmp*.nc tmp*.dump tmp*.tmp tmp_ngc.c ref_zarr_test_data.cdl
 

